### ISSUE TITLE
MD044 doesn't work for Markdown link text

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#369]
### rumdl version

0.1.10

### Installation method

mise

### Bug description

The MD044 rule (https://rumdl.dev/md044/) seems to work within YAML front matter (`title`), within regular Markdown text, but not when the text is in Markdown link text.

### Steps to reproduce

Given this Markdown file:

```markdown
---
title: test
---

test

[test](test.md)
```

And this `.rumdl.toml` file:

```toml
[MD044]
names = ["TEST"]
```

I'd expect three violations, but only get two:

```shell
test.md:2:8: [MD044] Proper name 'test' should be 'TEST' [*]
test.md:5:1: [MD044] Proper name 'test' should be 'TEST' [*]
```

### Expected behavior

There should also be a violation flagged at line 7.

### Actual behavior

There is no violation flagged at line 7.

### Configuration

```toml
[MD044]
names = ["TEST"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.11. MD044 now checks proper names in link text, image alt text, and WikiLinks.

Thanks for the report @eread!

**Comment by eread** [comment]
> Fixed in v0.1.11. MD044 now checks proper names in link text, image alt text, and WikiLinks.
> 
> Thanks for the report [@eread](https://github.com/eread)!

@rvben You're welcome! Thanks for building the tool!
